app/models/export_template.rb

```diff
@@ -19,6 +20,90 @@ def set_default_values
     end
   end
 
+  def zip?
+    kind == 'zip'
+  end
+
+  def tabular?
+    kind != 'zip'
+  end
+
+  def paths=(full_paths)
+    content["columns"] = full_paths.compact_blank
+      .map { [_1, full_path_hash(_1)] }
+      .filter { |_, current_column| current_column.present? }.map do |full_path, current_column|
+        libelle = case current_column[:source]
+        when 'tdc', 'repet'
+          active_type_de_champ = procedure.active_revision.types_de_champ.find_by(stable_id: current_column[:stable_id])
+          if active_type_de_champ
+            active_libelle = active_type_de_champ.libelle_for_path(current_column[:path])
+            libelle_for_path(full_path, active_libelle)
+          end
+        when 'dossier'
+          columns_meta[current_column[:path].to_sym][:libelle]
+        end
+
+        current_column.merge(libelle:) if libelle
+      end.compact
```

Comment:
ok, ces 2 lignes sont vachement subtiles :
- si un libelle est trouvé, alors l'itération du map renvoie le `current_column.merge(libelle:)`
- sinon l'itération renvoie `nil`

puis les nils sont retirés grace à l'appel final du `compact`.

Donc en fait on enregistre dans `content['columns']` que les paths dont on a pu trouver le libelle.

est ce qu'on pourrait réécrire le tout avec qq chose du genre

```
content['columns'] = json_paths
  .compact_map
  .map { JSON.parse(_1) }
  .map { _1[:libelle] = libelle_for_path_hash(_1) }
  .filter { _1[:libelle].present? }
```

en sortant la methode `libelle_for_path_hash` ?

Reply:
Joli :)
C'est par ici : https://github.com/demarches-simplifiees/demarches-simplifiees.fr/pull/10591/commits/d09afff1844769717fa67bf26c35abdc6b1468f5